generators/example/src/main/scala/TestHarness.scala

```diff
@@ -28,9 +38,8 @@ class TestHarness(implicit val p: Parameters) extends Module {
   // force Chisel to rename module
   override def desiredName = "TestHarness"
 
-  val dut = p(BuildTop)(clock, reset.toBool, p)
+  val dut = p(BuildTop)(clock, reset.toBool, p, io.success)
```

Comment:
Also a slight nit with the current approach: currently the `dut` instance is being renamed to `TestHarness` in some/all cases (I haven't tested extensively). We should add a `suggestName("dut")` here if that is still occurring.

Reply:
Wait why is this happening? `desiredName` should be changing the name of the Harness, not the dut.

Reply:
This has been happening for a while, I believe. I noticed it a few weeks ago and didn't really think much of it.

Reply:
![image](https://user-images.githubusercontent.com/8823803/70360646-f588bf80-1833-11ea-98e8-e2dc1dcf3268.png)

This is what I see currently... where the `TestHarness` is called `testHarness` (dictated by the `TestDriver.v`) and the instance of `Top` is called `TestHarness`. Frankly, I don't know why it is happening... but I haven't looked into it past what I posted here.

Reply:
It's entirely possible this is a weird chisel bug

Reply:
It's not coming from the `override def desiredName` line; I changed the name and `dut` is still named `TestHarness`.

Reply:
Interestingly `suggestName` also does not work. This seems like a @jackkoenig question.

Reply:
Yeah, I'm pretty sure `override def desiredName` doesn't work for an externally-included Blackbox verilog. I once solved this by putting my Verilog inline (using `setInline` where the first argument was `s"${this.desiredName}.v"`).

Reply:
@harrisonliew I'm not sure I follow- there's no blackbox here.

Reply:
Hmm it looks like this is still not resolved with the current PR. Have you tried addressing this? If not, I can try to ask some Chisel experts.

Reply:
This is outside my area of expertise, and I thought this was a pre-existing issue as well? Maybe we can resolve this in another PR

Reply:
Ok. Ill go ahead and make an issue.

Reply:
Ok found out two solutions. Figured I would put them here since it would be quick to add in this PR:

1. Call `suggestName` on the LazyModule
```
val lazytop = LazyModule(new Top()(p))
lazytop.suggestName("top")
val top = Module(lazytop.module)
```

2. Override the `suggestedName` in the `Top` module
```
override lazy val module = new TopModule(this)
override lazy val suggestedName = "sugN"
```

I personally think that the 1st option is a bit better since that is the proper way to override the default name (even though it is a bit more verbose).

Reply:
Fixed